src/rust/catcollar/futures/connect.rs

```diff
@@ -69,22 +71,34 @@ impl Future for ConnectFuture {
     /// Polls the underlying connect operation.
     fn poll(self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<Self::Output> {
         let self_: &mut ConnectFuture = self.get_mut();
-        match socket::connect(self_.fd as i32, &self_.addr) {
+        match unsafe {
+            libc::connect(
+                self_.fd,
+                (&self_.sockaddr as *const libc::sockaddr_in) as *const libc::sockaddr,
+                mem::size_of_val(&self_.sockaddr) as u32,
```

Comment:
While this will work, since sizeof sockaddr >= sizeof sockaddr_in, this really should be the latter and not the former.

Reply:
Nevermind, I see `sockaddr` is a `libc::sockaddr_in`, so this is ok.

Reply:
Thanks